app/models/katello/activation_key.rb

```diff
@@ -98,7 +98,11 @@ def products
       if cp_pools
         pools = cp_pools.collect { |cp_pool| Pool.find_by(:cp_id => cp_pool['id']) }
         pools.each do |pool|
-          all_products << pool.subscription.products
+          if pool.subscription
+            all_products << pool.subscription.products
+          else
+            Rails.logger.error("Pool #{pool.id} is missing its subscription id.")
```

Comment:
Well the whole purpose of this PR is to unblock the user.  Right now if a user hits this scenario, viewing the activation key list throws a 500 ISE and they can't do anything to remedy the situation.